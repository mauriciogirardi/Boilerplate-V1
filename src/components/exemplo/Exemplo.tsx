import * as S from './Exemplo.styles'

export function Exemplo({
  title = 'Boilerplate V1',
  description = 'TypeScript, ReactJS, NextJS and Styled Components'
}) {
  return (
    <S.ContainerExemplo>
      <S.Title>
        <S.Logo
          src="/img/icon-192.png"
          alt="Image of an atom and boilerplate version 1"
        />
        {title}
      </S.Title>
      <S.Description>{description}</S.Description>

      <S.Illustration
        src="/img/hero-illustration.svg"
        alt="Image of a man sitting in front of a computer"
      />
    </S.ContainerExemplo>
  )
}
