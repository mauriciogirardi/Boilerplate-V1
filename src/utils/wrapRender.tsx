import { render, RenderOptions } from '@testing-library/react'
import { ReactElement, ReactNode } from 'react'
import { ThemeProvider } from 'styled-components'
import { dark } from '../styles/themes/dark'

interface AllTheProvidersProps {
  children: ReactNode
}

const AllTheProviders = ({ children }: AllTheProvidersProps) => {
  return <ThemeProvider theme={dark}>{children}</ThemeProvider>
}

const wrapRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'

export { wrapRender as render }
