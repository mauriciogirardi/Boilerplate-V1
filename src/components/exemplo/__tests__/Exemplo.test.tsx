import { render, screen } from '../../../utils/wrapRender'
import { Exemplo } from '../Exemplo'

describe('<Exemplo/>', () => {
  it('should render correctly', () => {
    const { container } = render(<Exemplo />)

    expect(
      screen.getByRole('heading', { name: /Boilerplate V1/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    const { container } = render(<Exemplo />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092B;' })
  })
})
