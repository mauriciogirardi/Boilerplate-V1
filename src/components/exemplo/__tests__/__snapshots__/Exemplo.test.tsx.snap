// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Exemplo/> should render correctly 1`] = `
.c0 {
  background-color: #06092B;
  color: #FFFFFF;
  height: 100%;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  gap: 2rem;
  padding: 0 1.5rem;
}

.c1 {
  display: flex;
  align-items: center;
  gap: 1rem;
}

.c3 {
  font-weight: 400;
  font-size: 1rem;
}

.c2 {
  width: 5rem;
  height: 5rem;
}

.c4 {
  margin-top: 3rem;
  width: min(30rem,100%);
}

<main
  class="c0"
>
  <h1
    class="c1"
  >
    <img
      alt="Image of an atom and boilerplate version 1"
      class="c2"
      src="/img/icon-192.png"
    />
    Boilerplate V1
  </h1>
  <h2
    class="c3"
  >
    TypeScript, ReactJS, NextJS and Styled Components
  </h2>
  <img
    alt="Image of a man sitting in front of a computer"
    class="c4"
    src="/img/hero-illustration.svg"
  />
</main>
`;
